src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -542,20 +542,21 @@ else if (isSurrounded(ast) && unnecessaryParenAroundOperators(ast)) {
 
     /**
      * Tests if the given {@code DetailAST} is surrounded by parentheses.
-     * In short, does {@code ast} have a previous sibling whose type is
-     * {@code TokenTypes.LPAREN} and a next sibling whose type is {@code
-     * TokenTypes.RPAREN}.
      *
      * @param ast the {@code DetailAST} to check if it is surrounded by
      *        parentheses.
      * @return {@code true} if {@code ast} is surrounded by
      *         parentheses.
      */
     private static boolean isSurrounded(DetailAST ast) {
-        // if previous sibling is left parenthesis,
-        // next sibling can't be other than right parenthesis
         final DetailAST prev = ast.getPreviousSibling();
-        return prev != null && prev.getType() == TokenTypes.LPAREN;
+        final DetailAST parent = ast.getParent();
+        final boolean isPreviousSiblingLeftParenthesis = prev != null
+                && prev.getType() == TokenTypes.LPAREN;
+        final boolean isChildOfMethodCallWith = parent.getType() == TokenTypes.METHOD_CALL
```

Comment:
@Kevin222004 please do not leave us hanging... method call with what? :) `isChildOfMethodCallWith...` was just an example, how about `isChildOfMethodCallWithSiblingParenthesis`? or maybe even better is just `isMethodCallWithUnnecessaryParenthesis`?

Reply:
yes sorry my mistake I think `isMethodCallWithUnnecessaryParenthesis`

Reply:
done

Reply:
https://github.com/checkstyle/checkstyle/pull/12579#discussion_r1063619977 is it good